x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/FullTextFunction.java

```diff
@@ -73,19 +73,19 @@ public Expression query() {
     }
 
     /**
-     * Returns the resulting query as a String
+     * Returns the resulting query as an object
      *
-     * @return query expression as a string
+     * @return query expression as an object
      */
-    public final String queryAsText() {
+    public final Object queryAsObject() {
```

Comment:
Match query needs an object for it to work correctly on specific types. Changing the signature of this function to return an object